tests/tck-build-logic/src/main/groovy/org.graalvm.internal.tck-harness.gradle

```diff
@@ -102,6 +103,51 @@ def matrixDefault = [
 
 final String METADATA_GROUP = "Metadata"
 
+List<String> getNewerVersionsFor(String library, String startingVersion) {
+    def baseUrl = "https://repo1.maven.org/maven2"
+    String group = library.split(":")[0].replace(".", "/")
+    String artifact = library.split(":")[1]
+    def data = new URL(baseUrl + "/" + group + "/" + artifact + "/" + "maven-metadata.xml").getText()
+
+    return tck.getNewerVersionsFromLibraryIndex(data, startingVersion, library)
+}
+
+// gradle getExistingLibrariesWithNewVersions
+Provider<Task> getExistingLibrariesWithNewVersions = tasks.register("getExistingLibrariesWithNewVersions", DefaultTask) { task ->
+    task.setDescription("Returns list of all libraries coordinates")
+    task.setGroup(METADATA_GROUP)
+    task.doFirst {
+        Set<String> libraries = []
+        matchingCoordinates.forEach {
+            libraries.add(it.substring(0, it.lastIndexOf(":")))
+        }
+
+        List<String> newerVersions = new ArrayList<>()
+        libraries.forEach {
+            String libraryName = it
+            if (!libraryName.contains("samples") && !libraryName.contains("example")) {
```

Comment:
Hmm, where does this heuristic come from?

Reply:
We have tests for the repo infrastructure inside `samples` and `org.example` directories. They are only used to test repository functions like docker image pull, and simple test execution. Therefore, there is no point to search for any "version" of those tests.